### PR TITLE
feat: Teach the rust updater library about shorebird.yaml

### PR DESCRIPTION
**Pull request by eseidel**
The shorebird_init call now expects a string containing yaml contents in addition to the AppParameters struct.  AppParameters no longer contains product_id, client_id, update_url, or channel.  client_id is generated internally for the rust library and cached in the state cache.  product_id, base_url and channel are all read from shorebird.yaml which is passed in alongside the AppParameters.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: updater/cli/src/main.rs [hunk 1]
```diff
@@ -20,16 +20,17 @@ fn main() {
     let cli = Cli::parse();
 
     let config = updater::AppConfig {
-        client_id: "demo".to_string(),
```
Comment (felangel) [inline updater/cli/src/main.rs:23]: Was it intentional to remove `client_id` completely?

File: updater/library/src/config.rs [hunk 2]
```diff
@@ -59,27 +59,27 @@ impl ResolvedConfig {
...
         .to_owned();
     lock.cache_dir = config.cache_dir.to_string();
-    lock.client_id = config.client_id.to_string();
-    lock.product_id = config.product_id.to_string();
+    lock.product_id = yaml.product_id.to_string();
     lock.base_version = config.base_version.to_string();
     lock.original_libapp_path = config.original_libapp_path.to_string();
     lock.vm_path = config.vm_path.to_string();
     lock.is_initialized = true;
+    info!("Updater configured with: {:?}", lock);
```
Comment (felangel) [inline updater/library/src/config.rs:84]: do we want to keep this log in?

**Review by felangel — APPROVED** [review]
LGTM 🚀

File: updater/library/src/config.rs (hunk 2 above)
Comment (eseidel) [reply updater/library/src/config.rs:84]: Yes, that one was intentional.  Felt good to see what config the updater actually got.

**Pull request merged**